src/auth/src/credentials/service_account_credential.rs

```diff
@@ -48,6 +48,18 @@ pub(crate) struct ServiceAccountInfo {
     universe_domain: String,
 }
 
+impl std::fmt::Debug for ServiceAccountInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ServiceAccountInfo")
+            .field("client_email", &self.client_email)
+            .field("private_key_id", &"[censored]")
```

Comment:
I am not sure if you need to censor this.  I would hope that the ID of the key does not reveal anything about the key contents, or allow you to do anything nefarious.  And it would be useful to know what key was involved when troubleshooting something.

Maybe merge as-is and find out if it really needs censoring?

Reply:
I think you are right that the ID of the private key does not need to be censored. Fixed.